services/web/server/src/simcore_service_webserver/users/_users_repository.py

```diff
@@ -0,0 +1,422 @@
+import contextlib
+from typing import Any
+
+import simcore_postgres_database.errors as db_errors
+import sqlalchemy as sa
+from aiohttp import web
+from common_library.users_enums import UserRole
+from models_library.groups import GroupID
+from models_library.users import (
+    MyProfile,
+    UserBillingDetails,
+    UserID,
+    UserNameID,
+    UserPermission,
+)
+from pydantic import TypeAdapter, ValidationError
+from simcore_postgres_database.models.groups import groups, user_to_groups
+from simcore_postgres_database.models.products import products
+from simcore_postgres_database.models.users import UserStatus, users
+from simcore_postgres_database.models.users_details import (
+    users_pre_registration_details,
+)
+from simcore_postgres_database.utils_groups_extra_properties import (
+    GroupExtraPropertiesNotFoundError,
+    GroupExtraPropertiesRepo,
+)
+from simcore_postgres_database.utils_repos import (
+    pass_or_acquire_connection,
+    transaction_context,
+)
+from simcore_postgres_database.utils_users import (
+    UsersRepo,
+    generate_alternative_username,
+)
+from sqlalchemy import delete
+from sqlalchemy.engine.row import Row
+from sqlalchemy.ext.asyncio import AsyncConnection, AsyncEngine
+
+from ..db.plugin import get_asyncpg_engine
+from ._common.models import FullNameDict, ToUserUpdateDB
+from .exceptions import (
+    BillingDetailsNotFoundError,
+    UserNameDuplicateError,
+    UserNotFoundError,
+)
+
+_ALL = None
```

Comment:
I see no point in making an alias for None (Here I don't understand if you mean the type None or the value None, since it's treated specially).

I would prefer you either use it directly or just create something like a marker and use that to check.
```suggestion
_ALL = object()
```

Reply:
the intention here is to indicate that `None` is used as a sentinel that indicates "return all columns"
IN addition, your solution does not fit with the types in the annotations.

Reply:
The point is to document what `None` means in this context. 
Your solution does not work here because the input type do not include `object`
I refactored it.